Objects/typeobject.c

```diff
@@ -6841,6 +6841,26 @@ wrap_objobjargproc(PyObject *self, PyObject *args, void *wrapped)
     Py_RETURN_NONE;
 }
 
+static PyObject *
+wrap_cmpfunc(PyObject *self, PyObject *args, void *wrapped)
```

Comment:
Should this function emit a warning when it's called (to let the user know they're using something not supported in python3)?

Reply:
Good one, I have done the needful in commit [3d24c44](https://github.com/softdevteam/pygrate3/pull/7/commits/3d24c44ae9e9872faa660542ba8a3fb899093108)

Reply:
Will that warn every time you call `cmp` in the same source location, or just one warning per location?

Reply:
Will that warn every time you call `cmp` in the same source location, or just one warning per location?

Reply:
There are two areas to warn 1) Where cmp() is implemented i.e inside PyObject_Compare() and 2) At the __cmp__ implementation point where we attach the 'wrapfunc` logic to `TPSLOT('__cmp__'...`. So this was the last piece and only impacts `__cmp__` . The normal `cmp()` function already warns, the warning is implemented in:
`PyObject_Compare(PyObject *v, PyObject *w)`

Reply:
Formatting is wierd, bolded cmp == "__cmp__"